docs-configuration.md

```diff
@@ -0,0 +1,352 @@
+```
+
+output:
+
+```text
+dev config:
+{'my_project': 'I_am_dev'}
+my_secret: 123456
+```
+
+There are two important aspects here.
+**First**, by default, OS environment variable names
+must be prefixed with `bgq_`. You can change this prefix by setting
+the `environment_variables_prefix` parameter in the `Config` init method.
+
+**Second**, since secret properties don't exist in Python code
+they are resolved always lazily and only by name, using the `Config.resolve_property()` function.
+
+
+More interestingly, you can even resolve a current configuration name from OS environment:
+
+```python
+import os
+from bigflow import Config
+
+config = Config(name='dev',
+                properties={
+                   'my_project': 'I_am_{env}'
+                }).add_configuration(
+                name='prod',
+                properties={})
+
+os.environ['bgq_env'] = 'prod'
+config.pretty_print('')
+os.environ['bgq_env'] = 'dev'
+config.pretty_print('')
+```
+
+output:
+
+```text
+prod config:
+{'my_project': 'I_am_prod'}
+dev config:
+{'my_project': 'I_am_dev'}
+```
+ 
+## Dataset Config
+
+`DatasetConfig` is a convenient extension to `Config` designed for workflows which use `DatasetManager`
+to call Big Query SQL. 
+
+`DatasetConfig` defines four properties that are required by `DatasetManager`:
+
+* `project_id` &mdash; GCP project Id of an internal dataset.
+* `dataset_name` &mdash; Internal dataset name.
+* `internal_tables` &mdash; List of table names in an internal dataset. 
+  Fully qualified names of internal tables are resolved to `{project_id}.{dataset_name}.{table_name}`.  
+* `external_tables` &mdash; Dict that defines aliases for external table names.
+  Fully qualified names of external tables have to be declared explicitly.
+
+The distinction between internal and external tables shouldn't be treated too seriously.
+Internal means `mine`. External means any other. It's just a naming convention.
+
+For example:
+
+```python
+from biggerquery import DatasetConfig
+
+INTERNAL_TABLES = ['quality_metric']
+
+EXTERNAL_TABLES = {
+    'offer_ctr': 'not-my-project.offer_scorer.offer_ctr',
+    'box_ctr': 'other-project.box_scorer.box_ctr'
+}
+
+dataset_config = DatasetConfig(env='dev',
+                               project_id='my-project-dev',
+                               dataset_name='scorer',
+                               internal_tables=INTERNAL_TABLES,
+                               external_tables=EXTERNAL_TABLES
+                               )\
+            .add_configuration('prod',
+                               project_id='my-project-prod')
+``` 
+  
+Having that, a `DatasetManager` instance can be easily created:
+
+```python
+dataset_manager = dataset_config.create_dataset_manager()
+``` 
+
+Then, you can use short table names in SQL, `DatasetManager` resolves them to fully qualified names: 
+
+```python
+dataset_manager.collect('select * from {quality_metric}')
```

Comment:
Maybe some examples of fully resolved queries (e.g. `select * from my-project-dev. scorer. quality_metric` on dev)?